### PR TITLE
[Tests] Remove `$this` occurrences in future static data providers

### PR DESCRIPTION
**Pull request by alexandre-daubois**
| Q             | A
| ------------- | ---
| Branch?       | 5.4
| Bug fix?      | yes-ish
| New feature?  | no 
| Deprecations? | no 
| Tickets       | Easing https://github.com/symfony/symfony/pull/48668#issuecomment-1379892415
| License       | MIT
| Doc PR        | _NA_

### CODE REVIEW

File: src/Symfony/Component/Filesystem/Tests/PathTest.php [hunk 1]
```diff
@@ -589,7 +605,7 @@ public function testMakeAbsoluteDoesNotFailIfDifferentRoot(string $basePath, str
 
     public function provideMakeRelativeTests(): \Generator
     {
-        foreach ($this->providePathTests() as $set) {
+        foreach (static::PATH_TESTS as $set) {
```
Comment (OskarStark) [inline src/Symfony/Component/Filesystem/Tests/PathTest.php]: To use a constant as "provider" is quite uncommon. I would go with a static method instead, but no strong opinion

**Review by OskarStark — APPROVED** [review]

File: src/Symfony/Component/Filesystem/Tests/PathTest.php (hunk 1 above)
Comment (alexandre-daubois) [reply src/Symfony/Component/Filesystem/Tests/PathTest.php]: I don't have a strong opinion on this one either, but I prefer the static method way 👍 Updated!

**Comment by OskarStark** [comment]
Friendly ping @nicolas-grekas this is another PR on the road to static dataProviders. Thanks

**Review by nicolas-grekas — APPROVED** [review]

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @alexandre-daubois.

**Pull request merged**